token/cli/src/main.rs

```diff
@@ -876,6 +877,9 @@ async fn command_transfer(
         maybe_transfer_balance.unwrap()
     };
 
+    let maybe_fee =
+        ui_fee.map(|ui_amount| spl_token::ui_amount_to_amount(ui_amount, mint_info.decimals));
+
```

Comment:
do you think the fee calculations ever run into precision issues in practice? we convert here from a user-provided float string to a u64, and in the program do a somewhat tricky calculation with integer division and ceil, and `process_transfer` expects the values to match exactly...

im also not sure who in practice is going to be calculating out these values to pass into the cli. perhaps the user should pass in how many bps they expect the fee calculation to be, or they could pass a maximum amount and we calculate to ensure the actual is below that?

or the existing design may be fine, im just thinking out loud

Reply:
Your point is definitely valid. It could be that this is very rarely used, but at least it provides some additional optional protection.

Potentially as a safeguard later, we can add a `maximum-expected-fee` parameter to be more loose with checks.  Either way, I think there's value in both.